common/ts/__tests__/publish-task-test.ts

```diff
@@ -0,0 +1,119 @@
+  const result = await publishTask.getReportForTask(TASK_REPORT, METRICS, SQ_ENDPOINT, 999);
+
+  expect(Task.waitForTaskCompletion).toHaveBeenCalledWith(SQ_ENDPOINT, TASK_REPORT.ceTaskId, 999);
+  expect(Analysis.getAnalysis).toBeCalledWith(
+    { analysisId: '123', dashboardUrl: 'http://dashboardurl1', endpoint: SQ_ENDPOINT, metrics: { metrics: [] }, projectName: 'componentName' });
+
+  expect(result).toBe('dummy html');
+});
+
+it('get report string should return undefined if ceTask times out', async () => {
+  // Mock the ceTask timing out
+  jest.spyOn(Task, 'waitForTaskCompletion').mockImplementation(() => { throw new TimeOutReachedError(); });
+  jest.spyOn(Analysis, 'getAnalysis');
+  jest.spyOn(tl, 'warning').mockImplementation(() => null);
+
+  const result = await publishTask.getReportForTask(TASK_REPORT, METRICS, SQ_ENDPOINT, 999);
+
+  expect(result).toBeUndefined();
+  expect(Task.waitForTaskCompletion).toHaveBeenCalledWith(SQ_ENDPOINT, TASK_REPORT.ceTaskId, 999);
+  expect(tl.warning).toBeCalledWith('Task \'111\' takes too long to complete. Stopping after 999s of polling. No quality gate will be displayed on build result.');
+  expect(Analysis.getAnalysis).not.toBeCalled();
+});
+
+it('get report string should fail for non-timeout errors', async () => {
+  // Mock the ceTask timing out
+  jest.spyOn(Task, 'waitForTaskCompletion').mockImplementation(() => { throw new InvalidApiResourceVersionError('my error'); });
+  jest.spyOn(Analysis, 'getAnalysis');
+  jest.spyOn(tl, 'warning').mockImplementation(() => null);
+
+  expect.assertions(1);
+  try {
+    await publishTask.getReportForTask(TASK_REPORT, METRICS, SQ_ENDPOINT, 999);
+  } catch (e) {
+    expect(e).toEqual({ message: 'my error', name: 'Invalid resource version' });
+  }
+});
+
+it('task should not fail the task even if all ceTasks timeout', async () => {
+  const taskReport2 = new TaskReport({
+    ceTaskId: '222',
+    ceTaskUrl: 'http://taskurl2',
+    dashboardUrl: 'http://dashboardurl2',
+    projectKey: 'projectKey2',
+    serverUrl: 'http:/serverUrl2'
+  });
+
+  jest.spyOn(tl, 'getInput').mockImplementation(() => '1'); // set the timeout
+  jest.spyOn(tl, 'setResult')
+  jest.spyOn(tl, 'debug').mockImplementation(() => null);
+  jest.spyOn(tl, 'warning').mockImplementation(() => null);
+
+  tl.setVariable('SONARQUBE_SCANNER_PARAMS', 'anything...');
+  tl.setVariable('SONARQUBE_ENDPOINT', SC_ENDPOINT.toJson());
+
+  // Mock finding two report files to process
+  jest.spyOn(TaskReport, 'createTaskReportsFromFiles').mockImplementation(() => [TASK_REPORT, taskReport2]);
+
+  jest.spyOn(Task, 'waitForTaskCompletion').mockImplementation(() => { throw new TimeOutReachedError(); });
+  jest.spyOn(serverUtils, 'publishBuildSummary').mockImplementation(() => null);
+
+  await publishTask.default(EndpointType.SonarCloud);
+
+  expect(serverUtils.publishBuildSummary).toHaveBeenCalledTimes(1);
+  const publishSummaryMock = (serverUtils.publishBuildSummary as jest.Mock<any>);
```

Comment:
I think you can also store what's returned by the `spyOn` (that you call on line 106) to have the spy/mock instance of that method.